sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryElement.cs

```diff
@@ -0,0 +1,140 @@
+using System.Text.Json.Serialization;
+using System.Collections.Generic;
+
+namespace Azure.Core.Expressions.DataFactory
+{
+    /// <summary>
+    /// A class representing either a primitive value or an expression.
+    /// For details on DataFactoryExpressions see https://learn.microsoft.com/en-us/azure/data-factory/control-flow-expression-language-functions#expressions.
+    /// </summary>
+    /// <typeparam name="T"> Can be one of <see cref="string"/>, <see cref="bool"/>, <see cref="int"/>, <see cref="double"/>, <see cref="TimeSpan"/>,
+    /// <see cref="DateTimeOffset"/>, <see cref="Uri"/>, <see cref="IList{String}"/>, <see cref="IList{TElement}"/> where TElement has a <see cref="JsonConverter"/> defined,
+    /// or <see cref="IDictionary{String,String}"/>.</typeparam>
+#pragma warning disable SA1649 // File name should match first type name
+    [JsonConverter(typeof(DataFactoryElementJsonConverter))]
+    public sealed class DataFactoryElement<T>
+#pragma warning restore SA1649 // File name should match first type name
+    {
+        private readonly T? _literal;
+        private readonly DataFactoryElementKind _kind;
+        internal string? Value { get; }
+
+        /// <summary>
+        /// Initializes a new instance of DataFactoryExpression with a literal value.
+        /// </summary>
+        /// <param name="literal"> The literal value. </param>
+        public DataFactoryElement(T? literal) : this (literal, false)
+        {
+        }
+
+        internal DataFactoryElement(T? literal, bool asSecureString)
+        {
+            _kind = asSecureString ? DataFactoryElementKind.SecureString : DataFactoryElementKind.Literal;
+            _literal = literal;
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        public DataFactoryElementKind Kind => _kind;
+
+        // /// <summary>
+        // /// Gets whether this instance was constructed by a primitive value.
+        // /// </summary>
+        // public bool HasLiteral { get; }
+
+        /// <summary>
+        /// Gets the primitive value unless this instance is an expression.
+        /// </summary>
+        /// <exception cref="InvalidOperationException"> HasValue is false. </exception>
+        public T? Literal
+        {
+            get
+            {
+                if (_kind == DataFactoryElementKind.Literal)
+                    return _literal;
+                throw new InvalidOperationException("Cannot get value from non-literal.");
+            }
+        }
+
+        internal DataFactoryElement(string value, DataFactoryElementKind kind)
+        {
+            _kind = kind;
+            Value = value;
+        }
+
+        /// <inheritdoc/>
+        public override string? ToString()
+        {
+            if (_kind ==DataFactoryElementKind.Literal)
+                return _literal?.ToString();
+            if (_kind == DataFactoryElementKind.SecureString)
+                return new string('*', Value?.Length ?? 0);
```

Comment:
This isn't actually needed if the value was deserialized by the library. We can add a flag so that we only do the replace for input, or maybe don't do a replace at all for input.

Reply:
Thinking about it more, I think we should probably not modify the string on the client - for output values it should come back as asterisks from the service. For input values, the users is just saying they want the service to treat it as a secure string.